### PR TITLE
Machine Learning bugfixes

### PR DESCRIPTION
**Pull request by koenderks**
To start off, first commit should fix issues in https://github.com/jasp-stats/jasp-test-release/issues/266.

Commit requires additional packages `KneeArrower` plus dependencies (https://github.com/agentlans/KneeArrower)

### CODE REVIEW

**Comment by koenderks** [comment]
This should now also fix https://github.com/jasp-stats/jasp-test-release/issues/268

File: JASP-Engine/JASP/R/commonMachineLearningClustering.R [hunk 1]
```diff
@@ -20,7 +20,8 @@
   predictors <- predictors[predictors != ""]
   if (is.null(dataset))
     dataset <- .readDataSetToEnd(columns.as.numeric = predictors, exclude.na.listwise = predictors)
-
+  if(nrow(dataset) == 0 && length(unlist(options[["predictors"]])) > 0) # For some reason, .hasErrors() does not catch this error (it has to happen before scaling).
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/commonMachineLearningClustering.R]: @TimKDJ Can .hasErrors deal with NaN?

**Comment by koenderks** [comment]
I am working on some more fixes in the machine learning module right now, should I continue on this branch?

**Comment by JorisGoosen** [comment]
Well the PR is called "Machine Learning bugfixes" so that should be fine. However, if @AlexanderLyNL still needs to review it then having more changes might make this a bit akward?

**Comment by koenderks** [comment]
There will be small changes to quite a lot of files in what I am working on, so maybe it is better to merge this PR first?

**Comment by JorisGoosen** [comment]
I've just merged it, because any problems can be fixed in your next PR.

**Pull request merged**